openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -157,6 +160,17 @@ def update_account_settings(requesting_user, update, username=None):
     for serializer in user_serializer, legacy_profile_serializer:
         field_errors = add_serializer_errors(serializer, update, field_errors)
 
+    if "account_privacy" in update:
+        account_privacy = update["account_privacy"]
+        if account_privacy is not None:
+            try:
+                account_privacy_serializer = create_user_preference_serializer(
+                    existing_user, "account_privacy", account_privacy
+                )
+                validate_user_preference_serializer(account_privacy_serializer, "account_privacy", account_privacy)
+            except PreferenceValidationError as error:
+                field_errors["account_privacy"] = error.preference_errors["account_privacy"]
+
```

Comment:
Ignore the previous comment on this line.  I see that `account_privacy == None` and `'account_privacy' not in update` are treated differently below (at LL199-203)